pkg/expr/nodes.go

```diff
@@ -208,8 +189,8 @@ func (dn *DSNode) Execute(ctx context.Context, vars mathexp.Vars, s *Service) (m
 	pc := backend.PluginContext{
 		OrgID: dn.orgID,
 		DataSourceInstanceSettings: &backend.DataSourceInstanceSettings{
-			ID:  dn.datasourceID,
-			UID: dn.datasourceUID,
+			ID:  dn.datasource.Id,
```

Comment:
TODO? -- since we already have the datasoure model, can we fill in all the settings?

I presume that means somewhere else (not sure where) is filling up the data source info again